File: alligator/src/main/java/me/aartikov/alligator/listeners/DefaultNavigationErrorListener.java
package me.aartikov.alligator.listeners;

import android.support.annotation.NonNull;

import me.aartikov.alligator.exceptions.NavigationException;

/**
 * Date: 12.03.2017
 * Time: 15:16
 *
 * @author Artur Artikov
 */

/**
 * Default implementation of {@link NavigationErrorListener}. Wraps {@link NavigationException} to {@code RuntimeException} and throws it.
 */
public class DefaultNavigationErrorListener implements NavigationErrorListener {
	@Override
	public void onNavigationError(@NonNull NavigationException e) {
		throw new RuntimeException(e);
	}
}

File: alligator/src/main/java/me/aartikov/alligator/listeners/DefaultNavigationErrorListener.kt
package me.aartikov.alligator.listeners

import me.aartikov.alligator.exceptions.NavigationException

/**
 * Default implementation of [NavigationErrorListener]. Wraps [NavigationException] to `RuntimeException` and throws it.
 */
class DefaultNavigationErrorListener : NavigationErrorListener {
    override fun onNavigationError(e: NavigationException) {
        throw RuntimeException(e)
    }
}